beacon-light-client/plonky2/crates/circuits/src/deposit_accumulator_balance_aggregator_diva/first_level.rs

```diff
@@ -0,0 +1,506 @@
+        validator_status::{get_validator_relevance, get_validator_status},
+    },
+};
+use circuit::{circuit_builder_extensions::CircuitBuilderExtensions, Circuit, ToTargets};
+use circuit_derive::{CircuitTarget, SerdeCircuitTarget};
+use num::{BigUint, FromPrimitive};
+use plonky2::{
+    field::goldilocks_field::GoldilocksField,
+    hash::hash_types::HashOutTarget,
+    iop::target::BoolTarget,
+    plonk::{
+        circuit_builder::CircuitBuilder, circuit_data::CircuitConfig,
+        config::PoseidonGoldilocksConfig,
+    },
+};
+use plonky2_crypto::biguint::{BigUintTarget, CircuitBuilderBiguint};
+
+use crate::common_targets::{PoseidonMerkleBranchTarget, Sha256Target, ValidatorTarget};
+
+pub struct DepositAccumulatorBalanceAggregatorDivaFirstLevel {}
+
+#[derive(CircuitTarget, SerdeCircuitTarget)]
+#[serde(rename_all = "camelCase")]
+pub struct DepositAccumulatorBalanceAggregatorDivaFirstLevelTarget {
+    #[target(in)]
+    pub validator: ValidatorTarget,
+
+    #[target(in)]
+    pub validators_commitment_mapper_branch: PoseidonMerkleBranchTarget<24>,
+
+    #[target(in)]
+    #[serde(serialize_with = "biguint_to_str", deserialize_with = "parse_biguint")]
+    pub validator_gindex: BigUintTarget,
+
+    #[target(in)]
+    #[serde(with = "serde_bool_array_to_hex_string")]
+    pub deposit: PubkeyTarget,
+
+    #[target(in)]
+    #[serde(with = "serde_bool_array_to_hex_string")]
+    pub balance_leaf: Sha256Target,
+
+    #[target(in)]
+    #[serde(with = "serde_bool_array_to_hex_string_nested")]
+    pub balance_branch: Sha256MerkleBranchTarget<22>,
+
+    #[target(in)]
+    pub is_dummy: BoolTarget,
+
+    #[target(in, out)]
+    #[serde(serialize_with = "biguint_to_str", deserialize_with = "parse_biguint")]
+    pub current_epoch: BigUintTarget,
+
+    #[target(in, out)]
+    pub validators_commitment_mapper_root: HashOutTarget,
+
+    #[target(in, out)]
+    #[serde(with = "serde_bool_array_to_hex_string")]
+    pub balances_root: Sha256Target,
+
+    #[target(out)]
+    pub deposits_hash_tree_root: HashOutTarget,
+
+    #[target(out)]
+    pub accumulated_data: AccumulatedDataTargetDiva,
+}
+
+impl Circuit for DepositAccumulatorBalanceAggregatorDivaFirstLevel {
+    type F = GoldilocksField;
+    const D: usize = 2;
+    type C = PoseidonGoldilocksConfig;
+
+    const CIRCUIT_CONFIG: CircuitConfig = CircuitConfig::standard_recursion_config();
+
+    type Target = DepositAccumulatorBalanceAggregatorDivaFirstLevelTarget;
+
+    type Params = ();
+
+    fn define(
+        builder: &mut CircuitBuilder<Self::F, { Self::D }>,
+        _params: &Self::Params,
+    ) -> Self::Target {
+        let input = Self::read_circuit_input_target(builder);
+
+        let deposit_is_real = builder.not(input.is_dummy);
+
+        let deposits_hash_tree_root = HashOutTarget {
+            elements: poseidon(builder, input.deposit.to_targets())
+                .elements
+                .map(|x| builder.mul(x, deposit_is_real.target)),
+        };
```

Comment:
use `poseidon_or_zeroes(builder, input.deposit.to_targets(), deposit_is_real)` after we merge my branch